include/deal.II/base/numbers.h

```diff
@@ -37,11 +37,10 @@
 #define DEAL_II_HOST_DEVICE_ALWAYS_INLINE KOKKOS_FORCEINLINE_FUNCTION
 
 // clang++ assumes that all constexpr functions are __host__ __device__ when
-// compiling CUDA code, i.e, when Kokkos was configured with CUDA support.
-// This is problematic when calling non-constexpr functions in constexpr
-// functions. Hence, we need a way to annotate functions explicitly as
-// host-only.
-#if defined(__clang__) && defined(__CUDA__)
+// when Kokkos was configured with CUDA or HIP support. This is problematic
```

Comment:
```suggestion
// Kokkos was configured with CUDA or HIP support. This is problematic
```
Just minor documentation comment: duplicated from pervious line.